src/v/coproc/types.h

```diff
@@ -41,7 +41,7 @@ enum class enable_response_code : int8_t {
 enum class disable_response_code : int8_t {
     success = 0,
     internal_error,
-    script_id_does_not_exist,
+    script_id_does_not_exist = 7
```

Comment:
I don't understand how the subject

    error when casting *response_codes to coproc::errc codes

is related to the commit message which sounds more like a runtime issue

    coproc uses a catch all to print enable_/_disable_response_codes, so its an error to have an non unique id across all of the possible codes

but in any case the enum already provides unique values.